tower-balance/src/pool.rs

```diff
@@ -71,14 +73,17 @@ where
                     return Ok(Async::NotReady);
                 }
 
+                tracing::trace!("decided to add service to loaded pool");
                 try_ready!(self.maker.poll_ready());
+                tracing::trace!("making new service");
                 // TODO: it'd be great if we could avoid the clone here and use, say, &Target
                 self.making = Some(self.maker.make_service(self.target.clone()));
             }
         }
 
         if let Some(mut fut) = self.making.take() {
             if let Async::Ready(s) = fut.poll()? {
+                tracing::trace!("finished creating new service");
```

Comment:
Maybe we should record the number of services here as well?
```suggestion
                tracing::trace!(pool.services = self.services, message = "finished creating new service");
```